### DISCUSSION TITLE
Grpc Server implementation

### DISCUSSION THREAD

**Discussion by spiderko** [MarimerLLC/csla#3713] [Questions]
Hi all. I'm trying to introduce grpc server and client to our large project. Some of the business objects are having 200+ properties. Do I need to create proto files or do GrpcPortal have some magic which would allow me to avoid this pain?

**Comment by rockfordlhotka** [comment]
The data portal avoids that pain. The GrpcProxy works exactly like the HttpProxy, with a single endpoint on the server for all data portal calls.

If you encounter any issues let me know, as I don't think many people have used GrpcProxy yet, and although I've done basic testing, you might find edge cases.

**Reply by rockfordlhotka** [reply]
To be clear - if your code works today with HttpProxy, switching to GrpcProxy should require no code changes - only the configuration changes and setting up a GrpcPortal endpoint on the server.
